MaxText/checkpointing.py

```diff
@@ -49,27 +51,59 @@ def create_orbax_checkpoint_manager(
       options=CheckpointManagerOptions(
           create=True,
           save_interval_steps=save_interval_steps
-      )
+      ),
+      metadata={'iter': False}
   )
   max_logging.log("Checkpoint manager created!")
   return mngr
 
+def create_orbax_checkpoint_manager_pygrain(
+    checkpoint_dir: str,
+    enable_checkpointing: bool,
+    use_async: bool,
+    save_interval_steps: int,
+):
+  """Returns specified Orbax (async or not) CheckpointManager or None if checkpointing is disabled."""
+  if not enable_checkpointing:
+    max_logging.log("Checkpointing disabled, not creating checkpoint manager.")
+    return None
+  max_logging.log("Creating checkpoint manager (including data iterator)...")
+  p = epath.Path(checkpoint_dir)
+  if use_async:
+    checkpointer = {'default':AsyncCheckpointer(checkpoint.PyTreeCheckpointHandler()),
+                       'iter':Checkpointer(pygrain.PyGrainCheckpointHandler())}
+  else:
+    checkpointer = {'default':Checkpointer(checkpoint.PyTreeCheckpointHandler()),
+                       'iter':Checkpointer(pygrain.PyGrainCheckpointHandler())}
+
+  mngr = CheckpointManager(
+      p,
+      checkpointer,
+      options=CheckpointManagerOptions(
+          create=True,
+          save_interval_steps=save_interval_steps
+      ),
+      metadata={'iter': True}
+  )
+  max_logging.log("Checkpoint manager created!")
+  return mngr
 
 def load_state_if_possible(checkpoint_manager: CheckpointManager,
-                           first_checkpoint_path: str,
+                           load_parameters_path: str,
                            load_from_other_directory: str,
                            load_from_other_directory_step: int,
                            abstract_unboxed_pre_state: train_state.TrainState,
+                           load_data_iterator_from_checkpoint,
+                           iterator,
                            mesh,
                            state_mesh_annotations):
   """Loads TrainState as possible from the inputs.
 
   Args:
     checkpoint_manager: if the checkpoint_manager has a valid checkpoint, return
       that TrainState. This enables a full reload of a run in progress.
-    first_checkpoint_path: if there is no checkpoint in the checkpoint manager,
-      return the Params from the first_checkpoint_path if they exist. This
-      enables loading just the parameters and is intended for finetuning.
+    load_parameters_path: This enables loading just the parameters and is intended 
```

Comment:
Thanks for fixing this name. It was driving me crazy the last time I read the code.